src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -737,15 +736,37 @@ AttributeEquivalenceClassForEquivalenceClass(EquivalenceClass *plannerEqClass,
 
 		if (IsA(strippedEquivalenceExpr, Param))
 		{
-			List *parentParamList = relationRestriction->parentPlannerParamList;
+			PlannerInfo *outerNodeRoot = relationRestriction->plannerInfo->parent_root;
+			ListCell *outerPlanParamListCell = NULL;
 			Param *equivalenceParam = (Param *) strippedEquivalenceExpr;
 
-			expressionVar = GetVarFromAssignedParam(parentParamList,
-													equivalenceParam);
+			/*
+			 * A Param can reference a Var in any of the outer plan nodes. Search
+			 * them until we find the corresponding Var.
+			 */
+			foreach(outerPlanParamListCell, relationRestriction->outerPlannerParamLists)
```

Comment:
I'd be in favor of moving this logic to a separate function, especially if you agree with my below comment.